include/engine/routing_algorithms/routing_base.hpp

```diff
@@ -250,81 +251,83 @@ template <class DataFacadeT, class Derived> class BasicRoutingInterface
-
-                  unpacked_path.back().entry_classid = facade.GetEntryClassID(edge_data.id);
-                  unpacked_path.back().turn_instruction = turn_instruction;
-                  unpacked_path.back().duration_until_turn += (edge_data.distance - total_weight);
-              });
-
-          std::size_t start_index = 0, end_index = 0;
-          std::vector<unsigned> id_vector;
-          std::vector<EdgeWeight> weight_vector;
-          std::vector<DatasourceID> datasource_vector;
-          const bool is_local_path = (phantom_node_pair.source_phantom.packed_geometry_id ==
-                                      phantom_node_pair.target_phantom.packed_geometry_id) &&
-                                     unpacked_path.empty();
-
-          if (target_traversed_in_reverse)
-          {
-              id_vector = facade.GetUncompressedReverseGeometry(
-                  phantom_node_pair.target_phantom.packed_geometry_id);
-
-              weight_vector = facade.GetUncompressedReverseWeights(
-                  phantom_node_pair.target_phantom.packed_geometry_id);
-
-              datasource_vector = facade.GetUncompressedReverseDatasources(
-                  phantom_node_pair.target_phantom.packed_geometry_id);
+                const auto total_weight =
+                    std::accumulate(weight_vector.begin(), weight_vector.end(), 0);
+
+                BOOST_ASSERT(weight_vector.size() == id_vector.size() - 1);
+                const bool is_first_segment = unpacked_path.empty();
+
+                const std::size_t start_index =
+                    (is_first_segment
+                         ? ((start_traversed_in_reverse)
+                                ? weight_vector.size() -
+                                      phantom_node_pair.source_phantom.fwd_segment_position - 1
+                                : phantom_node_pair.source_phantom.fwd_segment_position)
+                         : 0);
+                const std::size_t end_index = weight_vector.size();
+
+                BOOST_ASSERT(start_index >= 0);
+                BOOST_ASSERT(start_index < end_index);
+                for (std::size_t segment_idx = start_index; segment_idx < end_index; ++segment_idx)
+                {
+                    unpacked_path.push_back(
+                        PathData{id_vector[segment_idx + 1],
+                                 name_index,
+                                 weight_vector[segment_idx],
+                                 extractor::guidance::TurnInstruction::NO_TURN(),
+                                 {{0, INVALID_LANEID}, INVALID_LANE_DESCRIPTIONID},
+                                 travel_mode,
+                                 INVALID_ENTRY_CLASSID,
+                                 datasource_vector[segment_idx],
+                                 util::guidance::TurnBearing(0),
+                                 util::guidance::TurnBearing(0)});
```

Comment:
the rest of this part was not aligned properly. The addition of the turn-bearings is the only change here.